### ISSUE TITLE
Hyas Build Failure with Hugo v0.123.4 on Darwin/ARM64 Architecture

### ISSUE THREAD

**Issue by kevinjcharles** [gethyas/hyas#460]
Dear Hyas Support Team,

## Description

I am experiencing an issue with the Hyas Hugo framework when trying to build with Hugo version 0.123.4. The build process initiates correctly, but eventually fails during the rendering phase, particularly with the execution of templates. The error seems to stem from a potential infinite recursion or timeout in the partial templates, particularly `head/head.html` and `head/stylesheet.html`.

### Steps to Reproduce

1. Install Hugo version 0.123.4 on a macOS with darwin/arm64 architecture.
2. Set up a new Hyas project or use an existing one.
3. Run the command `npm run dev` to start the development server.

Unfortunately, I do not have a demo project readily available to link, but the issue occurs with standard Hyas project configurations.

### Expected Result

The expected result is a successful build and rendering of pages without any errors, allowing the local development server to run smoothly.

### Actual Result

The build process fails with an error message indicating a timeout and potential infinite recursion in the template execution. The specific error messages are:

- `ERROR render of "page" failed: execute of template failed: template: _default/single.html: error calling partial: partial "head/head.html" timed out after 30s.`
- `Error building site: render: failed to render pages: execute of template failed: template: index.html: error calling partial: error calling Permalink: not supported.`

### Environment

- Hugo Version: 0.123.4
- Extended: Yes
- OS: darwin
- Architecture: arm64
- Build Date: 2024-02-26T16:33:05Z
- Vendor Info: gohugoio

Paste the information here as shown by `npm run info`

npm run info

> test-hyas-project@0.0.0 preinfo
> npm version

{
  'test-hyas-project': '0.0.0',
  npm: '10.4.0',
  node: '21.6.1',
  acorn: '8.11.3',
  ada: '2.7.4',
  ares: '1.20.1',
  base64: '0.5.1',
  brotli: '1.1.0',
  cjs_module_lexer: '1.2.2',
  cldr: '44.0',
  icu: '74.1',
  llhttp: '9.1.3',
  modules: '120',
  napi: '9',
  nghttp2: '1.58.0',
  nghttp3: '0.7.0',
  ngtcp2: '0.8.1',
  openssl: '3.0.12+quic',
  simdjson: '3.6.3',
  simdutf: '4.0.8',
  tz: '2023c',
  undici: '5.28.2',
  unicode: '15.1',
  uv: '1.47.0',
  uvwasi: '0.0.19',
  v8: '11.8.172.17-node.19',
  zlib: '1.3.0.1-motley-40e35a7'
}

> test-hyas-project@0.0.0 info
> npm list

test-hyas-project@0.0.0 /Users/kevin/Sites/test-hyas-project
├── @hyas/bootstrap@1.0.4
├── @hyas/images@2.0.3
├── @hyas/seo@2.1.0
├── exec-bin@1.0.0
├── gethyas@2.2.2
├── hugo-installer@4.0.1
└── shx@0.3.4

> test-hyas-project@0.0.0 postinfo
> exec-bin node_modules/.bin/hugo/hugo version

hugo v0.123.4-21a41003c4633b142ac565c52da22924dc30637a+extended darwin/arm64 BuildDate=2024-02-26T16:33:05Z VendorInfo=gohugoio

**Comment by h-enk** [comment]
Thanks for reporting/ confirmed (on windows/amd64).

The latest working release is [v0.122.0](https://github.com/gohugoio/hugo/releases/tag/v0.122.0).

I will be looking into this — see also [v0.123.0](https://github.com/gohugoio/hugo/releases/tag/v0.123.0)

**Comment by h-enk** [comment]
OK, I got the cause — it's because of two files (with the same name) mounted twice (resulting in two conflicts). To be fixed!

Quick fix for now, add to `config/_default/module.toml` the following `excludeFiles`:

```toml
## assets
[[mounts]]
  source = "node_modules/@hyas/core/assets"
  target = "assets"
  excludeFiles = ["/scss/app.scss", "/js/app.js"]

[[mounts]]
  source = "assets"
  target = "assets"
```

**Comment by h-enk** [comment]
Quick fix for Hugo v0.123.6 (we're getting there):

```toml
## assets
[[mounts]]
  source = "node_modules/@hyas/core/assets"
  target = "assets"
  excludeFiles = ["scss/app.scss"]

[[mounts]]
  source = "assets"
  target = "assets"
```

**Comment by h-enk** [comment]
https://github.com/gohugoio/hugo/issues/12175

**Comment by h-enk** [comment]
Fixed w/ [v123.7](https://github.com/gohugoio/hugo/releases/tag/v0.123.7)
